dataactvalidator/config/sqlRules.csv

```diff
@@ -1,7 +1,14 @@
 "SELECT af.row_number, af.bydirectreimbursablefundingsource FROM award_financial AS af WHERE af.submission_id = {} AND COALESCE(lower(af.bydirectreimbursablefundingsource),'') NOT IN ('', 'r', 'd')",B12,"Reimbursable flag indicator is required when reporting obligation or outlay USSGL account balances (excluding downward adjustments SGL accounts). Valid values are ""R"" for ""ReimbursableFundingSource"" and ""D"" for ""DirectFundingSource"".","The reimbursable flag indicator (bydirectreimbursablefundingsource) must be empty, R, or D",n,award_financial,fatal
 "SELECT op.row_number, op.bydirectreimbursablefundingsource FROM object_class_program_activity AS op WHERE op.submission_id = {} AND COALESCE(lower(op.bydirectreimbursablefundingsource),'') NOT IN ('', 'r', 'd')",B12,"Reimbursable flag indicator is required when reporting obligation or outlay USSGL account balances (excluding downward adjustments SGL accounts). Valid values are ""R"" for ""ReimbursableFundingSource"" and ""D"" for ""DirectFundingSource"".","The reimbursable flag indicator (bydirectreimbursablefundingsource) must be empty, R, or D",n,program_activity,fatal
+"SELECT af.row_number, af.objectclass from award_financial AS af WHERE af.submission_id = {} AND af.objectclass NOT IN (SELECT object_class_code FROM object_class)",B11,"Must be valid 3-digit object class as defined in OMB Circular A-11 Section 83.6, or a 4-digit code which includes a 1-digit prefix that distinguishes direct, reimbursable, and allocation obligations. Do not include decimal points when reporting in the Schema.","Failed rule: Must be valid 3-digit object class as defined in OMB Circular A-11 Section 83.6, or a 4-digit code which includes a 1-digit prefix that distinguishes direct, reimbursable, and allocation obligations. Do not include decimal points when reporting in the Schema.",n,award_financial,warning
+"SELECT af.row_number, af.objectclass, af.bydirectreimbursablefundingsource FROM award_financial af WHERE af.submission_id = {} AND LENGTH(af.objectclass) = 4 AND ((LEFT(af.objectclass,1) = '1' AND lower(bydirectreimbursablefundingsource) IS DISTINCT FROM 'd') OR (LEFT(af.objectclass,1) = '2' AND lower(af.bydirectreimbursablefundingsource) IS DISTINCT FROM 'r'))",B11,"Must be valid 3-digit object class as defined in OMB Circular A-11 Section 83.6, or a 4-digit code which includes a 1-digit prefix that distinguishes direct, reimbursable, and allocation obligations. Do not include decimal points when reporting in the Schema.","Failed rule: If object class is 4 digits, check that 1 digit prefix is consistent with value for ByDirectReimbursableFundingSource",n,award_financial,warning
```

Comment:
I'm guessing one of the pairs for B11 is supposed to be for program activity instead of both being award financial?  This should cut the 8 extra errors in half, leaving 4 unexpected.  Given that this doesn't seem to include a removal of the existing B11 rule, removing those 6 errors will actually leave us two short.  One of those is because the prefix rule is passing anything other than '1' or '2', so 'ABCD' passes the prefix rule.  The other is because this rule is exactly 4 whereas the old implementation was > 3, so '25412' doesn't get checked.  Only checking for exactly 4 makes sense, so we can drop the expected errors by one, I'm not sure what the behavior should be for unexpected prefixes, do we show both errors or just the error of not being a valid object class (until there's some weird exception).